nexus_constructor/add_component_window.py

```diff
@@ -122,6 +126,28 @@ def setupUi(self, parent_dialog):
         # Validate the default values set by the UI
         self.unitsLineEdit.validator().validate(self.unitsLineEdit.text(), 0)
         self.nameLineEdit.validator().validate(self.nameLineEdit.text(), 0)
+        self.addFieldPushButton.clicked.connect(self.add_field)
+        self.removeFieldPushButton.clicked.connect(self.remove_field)
+
+        # Set whatever the default nx_class is so the fields autocompleter can use the possible fields in the nx_class
+        self.on_nx_class_changed()
+
+        self.fieldsListWidget.itemClicked.connect(self.select_field)
+
+    def add_field(self):
+        item = QListWidgetItem()
+        field = FieldWidget(self.possible_fields, self.fieldsListWidget)
+        field.something_clicked.connect(partial(self.select_field, item))
+        item.setSizeHint(field.sizeHint())
+        self.fieldsListWidget.addItem(item)
+        self.fieldsListWidget.setItemWidget(item, field)
+
+    def select_field(self, widget):
+        self.selected_widget = widget
+        self.fieldsListWidget.setItemSelected(widget, True)
+
+    def remove_field(self):
+        self.fieldsListWidget.takeItem(self.fieldsListWidget.row(self.selected_widget))
```

Comment:
If you use:
```python
selected_row = self.fieldsListWidget.row(self.fieldsListWidget.currentItem())
self.fieldsListWidget.takeItem(selected_row)
```
You will no longer need to hold `self.selected_widget` in this class

Reply:
I have no idea why, but i can't get that to work. `currentRow()` or `currentItem()` seem to never change even if the item is selected.

Reply:
I've sorted it now by using `selectedItems()`